src/components/PositionMarker.js

```diff
@@ -0,0 +1,28 @@
+import { Marker, useMapEvents } from "react-leaflet";
+
+
+const PositionMarker = ({ location, setLocation, markerLocation, setMarkerLocation }) => {
+  useMapEvents({
+    click(e) {
+      if (setLocation && setMarkerLocation) {
+        setMarkerLocation(e.latlng);
+        setLocation(`${e.latlng.lat},${e.latlng.lng}`);
+      }
+    }
+  });
+
+  if (location && typeof location === "string") {
```

Comment:
I would recommend to change location to an object:
`{lat: value, lon:value}`
otherwise you have to remember which value comes first.

Reply:
I agree. I will change that in the API and then in the frontend